packages/core/src/service/services/product-variant.service.ts

```diff
@@ -130,8 +131,21 @@ export class ProductVariantService {
             });
     }
 
-    getVariantsByProductId(ctx: RequestContext, productId: ID): Promise<Array<Translated<ProductVariant>>> {
-        const qb = this.connection.getRepository(ctx, ProductVariant).createQueryBuilder('productVariant');
+    getVariantsByProductId(
+        ctx: RequestContext,
+        productId: ID,
+        options: ListQueryOptions<ProductVariant> = {},
+    ): Promise<PaginatedList<Translated<ProductVariant>>> {
+        const rawConnection = this.connection.rawConnection;
+        const qb = this.connection.getRepository(ctx, ProductVariant).createQueryBuilder('productvariant');
+
+        const filter = parseFilterParams(rawConnection, ProductVariant, options.filter);
```

Comment:
I notice that quite a significant amount of logic here is being duplicated from the `ListQueryBuilder.build()` method. I am wondering whether it would be possible to use the ListQueryBuilder, and then add the required clauses (ie. `.andWhere('productvariant.productId = :productId', { productId })` to the returned SelectQueryBuilder object.

Did you try that approach yet?

Reply:
Hello! Thank you! I agree that would be better to use `ListQueryBuilder.build()` here. I will do it.